### DISCUSSION TITLE
GPG signatures in term window fail due to "Screen or window to small."

### DISCUSSION THREAD

**Discussion by briandailey** [ghostty-org/ghostty#4221] [Q&A] [answered]
When I try to sign a git commit, I see the following:

```
error: gpg failed to sign the data:
[GNUPG:] KEY_CONSIDERED [..snip..] 2
[GNUPG:] BEGIN_SIGNING H8
[GNUPG:] PINENTRY_LAUNCHED 2304 curses 1.3.1 /dev/ttys032 xterm-ghostty - 20620/501/4 501/20 0
gpg: signing failed: Screen or window too small
[GNUPG:] FAILURE sign 83886381
gpg: signing failed: Screen or window too small

fatal: failed to write commit object
```

This works fine in Terminal or iTerm. I assume there is something about how the window size is reported in Ghostty.

This is on MacOS 15.1.1

**Comment by mitchellh** [comment, marked as answer]
This is a terminfo issue. Your GPG agent needs to be restarted from Ghostty: https://ghostty.org/docs/help/terminfo

**Reply by jackbravo** [reply]
you restart the agent by killing it first with: `gpgconf --kill gpg-agent`. Then next command should succeed.

**Reply by jiacai2050** [reply]
Why the terminal.app works out of box?

**Reply by 00-kat** [reply]
I believe the problem is that Terminal.app's terminfo entry is in the global database (since it's either a generic one, or it's because Terminal.app is older), while Ghostty's one is only available from inside Ghostty (because of some specific environment variables Ghostty sets to make it available), so the GPG agent doesn't know of Ghostty's terminfo unless you restart it from inside Ghostty.

**Reply by jiacai2050** [reply]
Which software cause this issue? [pinentry](https://www.gnupg.org/related_software/pinentry/) or something else?

**Reply by 00-kat** [reply]
Not sure, but it's probably pinentry based on OP's logs.

**Comment by deadlysyn** [comment]
for me, i just needed `export GPG_TTY=$(tty)`
